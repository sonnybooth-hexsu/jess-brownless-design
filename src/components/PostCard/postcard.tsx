import React from "react"
import { Link } from "gatsby"

type PostCardProps = {
  title: string
  category: string
  path: string
  image: string
  date: string
}

export const PostCard = ({
  title,
  category,
  path,
  image,
  date,
}: PostCardProps) => {
  const d = new Date(date)
  const ye = new Intl.DateTimeFormat("en", { year: "numeric" }).format(d)
  const mo = new Intl.DateTimeFormat("en", { month: "short" }).format(d)

  if (category == "Aesthetics") {
  }

  if (category == "Eye Surgery") {
  }

  if (category === "Eyes") {
  }

  return (
    <Link to={path} className="col-span-12 md:col-span-4">
      <div className="">
        <img
          className="h-full col-span-12"
          loading="lazy"
          width="100%"
          src={image}
          alt="Post"
        />
      </div>

      <div className="inline text-sm text-Brand-green">
        <span className="overline">{category}</span>
      </div>
      <h2 data-testid="postcardTitle" className="mt-4 displayLG lg:displayXL">
        {title}
      </h2>
      <div className="lg:pr-12 lg:pl-6 lg:py-12 lg:col-span-5">
        <div className="py-1 text-center uppercase">{mo}</div>
        <div className="py-3 text-center">
          <span className="mt-1">{ye}</span>
        </div>
      </div>
    </Link>
  )
}
