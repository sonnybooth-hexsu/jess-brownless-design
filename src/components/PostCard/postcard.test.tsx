/* eslint no-undef: 0 */

import React from "react"
import { render } from "@testing-library/react"
import { getByText, getByTestId } from "@testing-library/dom"
import { PostCard } from "./postcard"

describe("<PostCard />", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <PostCard
        title="Post card title"
        category="Post card category"
        path="/test"
        date="12/01/98"
        image="/test-image.jpg"
      />
    )

    expect(container).toMatchSnapshot()
  })

  it("should recieve title prop", () => {
    const { container } = render(
      <PostCard
        title="Post card title"
        category="Post card category"
        path="/test"
        date="12/01/98"
        image="/test-image.jpg"
      />
    )

    const titleEl = getByTestId(container, "postcardTitle")

    expect(getByText(titleEl, "Post card title"))
  })
})
