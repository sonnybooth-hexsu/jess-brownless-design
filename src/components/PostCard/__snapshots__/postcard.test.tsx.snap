// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div>
  <a
    href="/test"
  >
    <div
      class="grid grid-cols-12 row-gap-0 col-gap-6 mb-8"
    >
      <div
        class="hidden w-24 col-span-12 mb-8 md:inline-block md:mb-0 md:w-auto md:col-span-2 lg:col-span-1"
      >
        <div
          class="py-1 text-center text-white uppercase bg-Brand-green"
        >
          Dec
        </div>
        <div
          class="py-3 text-center bg-whiteBright"
        >
          <span
            class="block mt-1 text-grayWarm"
          >
            1998
          </span>
        </div>
      </div>
      <div
        class="col-span-12 shadow-sm bg-whiteBright md:col-span-10 lg:col-span-11"
      >
        <div
          class="flex flex-col lg:grid lg:grid-cols-11 lg:gap-6"
        >
          <div
            class="lg:col-span-6"
          >
            <img
              alt="Post"
              class="h-full"
              loading="lazy"
              src="/test-image.jpg"
              width="100%"
            />
          </div>
          <div
            class="p-10 pb-12 lg:pr-12 lg:pl-6 lg:py-12 lg:col-span-5"
          >
            <div
              class="inline-block px-4 py-1 text-sm text-white rounded-3xl bg-Brand-green"
            >
              <div
                class="flex items-center"
              >
                <img
                  alt="Category"
                  class="w-6 h-6 mr-2"
                  src="test-file-stub"
                />
                <span
                  class="font-light tracking-wider uppercase"
                >
                  Post card category
                </span>
              </div>
            </div>
            <h3
              class="mt-4 displayLG lg:displayXL"
              data-testid="postcardTitle"
            >
              Post card title
            </h3>
          </div>
        </div>
      </div>
    </div>
  </a>
</div>
`;
